### DISCUSSION TITLE
Some markdown not rendering

### DISCUSSION THREAD

**Discussion by ab492** [spatie/laravel-markdown#78] [Q&A]
Hi all. I've got a web app that uses Laravel Markdown and I've got some strange behaviour I'm trying to debug. When I write markdown directly in my blade view like this (screenshot as the backticks were messing with Github's markdown rendering):

![Screenshot 2024-06-27 at 07 28 00](https://github.com/spatie/laravel-markdown/assets/35888770/758c6dca-00f8-490c-8b11-6e8eaab62373)

It renders perfectly, like this:

<img width="246" alt="Screenshot 2024-06-27 at 07 24 41" src="https://github.com/spatie/laravel-markdown/assets/35888770/7ee30cfb-0833-4178-b50e-7023c11cc007">

However, when I try to use a controller with the view and pass the markdown as a property to the `@markdown` directive, it doesn't render correctly.

**Controller:**

```
<?php

namespace App\Http\Controllers;

use Illuminate\Http\Request;

class MarkdownController extends Controller
{
    public function show()
    {
        $markdownContent = "
        # My title

        This is a [link to our website](https://spatie.be)

        ```php
        echo 'Hello world';
        ```
        ";

        return view('markdown.show', ['markdownContent' => $markdownContent]);
    }
}
```

**View:**
```
@markdown($markdownContent)
```

**Incorrect output:**

<img width="439" alt="Screenshot 2024-06-27 at 07 30 42" src="https://github.com/spatie/laravel-markdown/assets/35888770/6711ee02-aa3b-484d-9242-e6e8032c1b70">

Does anyone have any suggestions for why one might work and the other doesn't please?

**Comment by matfire** [comment]
don't know the exact answer, but it could be laravel's xss protection ?
> Note: If you're outputting the Markdown through blade, rather than direct input, you will need to use the unescaped blade statement to prevent Laravel's XSS protection stripping the tags:

```blade
{!! $article->content !!}
```
mentioned here https://spatie.be/docs/laravel-markdown/v1/usage-in-blade/general-usage

Note: this works for me if I use it with the `x-markdown` component; haven't tested it with the `@markdown`
